Extension/src/LanguageServer/configurations.ts

```diff
@@ -117,32 +118,7 @@ export class CppProperties {
         let rootPath: string = rootUri ? rootUri.fsPath : "";
         this.currentConfigurationIndex = new PersistentFolderState<number>("CppProperties.currentConfigurationIndex", -1, rootPath);
         this.configFolder = path.join(rootPath, ".vscode");
-
-        let configFilePath: string = path.join(this.configFolder, "c_cpp_properties.json");
```

Comment:
I moved this because it was causing the JSON to be parsed 2 extra times unnecessarily.

Reply:
Can you double check to make sure that a bogus config doesn't get sent to the extension while we wait for the compiler defaults to arrive? I seem to recall adding this to the constructor for a reason.

Reply:
I believe this change requires my other pull request to be 100% correct during the initial load: https://github.com/Microsoft/vscode-cpptools/pull/3178 . Was there something else you wanted me to double check in particular? I tested compile_commands and configurationProvider scenarios. The handleConfigurationChange() method checks for undefined now (i.e. before it's set), which can occur with the async buildVcpkgIncludePath -- that might be what you're referring to?

Reply:
I approved that PR

Reply:
I double checked and tested it and it seems good to me.